wordpress-mobile/WordPress-iOS
# Replace WPSearchController with UISearchController

Fixes #4343 and #3747.

I've _finally_ managed to replace `WPSearchController` with `UISearchController` :tada:

We use search in 3 places in the app, which I've updated:

* `BlogListViewController`: This is the simplest implementation, as it's just a `tableView` with the search bar in the header.
* `PostListViewController` and `PageListViewController` is slightly more complex due to the table view's header also containing a filter bar in the posts list (when a blog has multiple authors). I managed to get this working by wrapping them both in a stack view.
* `ThemeBrowserViewController`: This was the most complex, due to the view controller using a collection view. I tried placing the search bar into the existing section header, but the system-provided presentation animation didn't work so well with that setup. I managed to get it working by putting the search bar into its own separate section header.

The setup is actually relatively straightforward for each one (it was mainly a case of working out _how_ to get it to work in the first place). In addition to inserting the search controller's search bar into the view hierarchy and updating search results along with the search terms, each of the view controllers in question implements keyboard handling (`will`/`did`/`show`/`hide`) to ensure that that their content and scroll insets are correct.

To test:

* Ensure that the search functionality works as expected in each of the areas mentioned above.
* Ensure that scroll and content insets are correct when search is activated, deactivated, and when the keyboard is hidden and shown (also test iPad, as it has only the smaller quick type bar visible when an external keyboard is connected).
* Check that the search bar resizes correctly between portrait and landscape, different device sizes, and multitasking.
* Ensure that the visual appearance looks right.

Let me know if you have any questions!

Needs review: @kurzee 

@jleandroperez You mentioned you might be interested in this, so feel free to take a look if you like!
@mattmiklic Would you might checking the appearance of the new search bar?
